client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * External dependencies
+ */
+import React, {
+	useCallback,
+	useContext,
+	useEffect,
+	useMemo,
+	useState,
+} from 'react';
+import { __ } from '@wordpress/i18n';
+import { Button, Card, CardBody } from '@wordpress/components';
+import interpolateComponents from 'interpolate-components';
+
+/**
+ * Internal dependencies
+ */
+import WizardTaskContext from '../wizard/task/context';
+import CollapsibleBody from '../wizard/collapsible-body';
+import WizardTaskItem from '../wizard/task-item';
+import {
+	useEnabledPaymentMethodIds,
+	useGetAvailablePaymentMethodIds,
+	useSettings,
+} from '../../data';
+import PaymentMethodCheckboxes from '../../components/payment-methods-checkboxes';
+import PaymentMethodCheckbox from '../../components/payment-methods-checkboxes/payment-method-checkbox';
+import { LoadableBlock } from '../../components/loadable';
+import LoadableSettingsSection from '../../settings/loadable-settings-section';
+import CurrencyInformationForMethods from '../../components/currency-information-for-methods';
+
+const usePaymentMethodsCheckboxState = () => {
+	const availablePaymentMethods = useGetAvailablePaymentMethodIds();
+	const [ paymentMethodsState, setPaymentMethodsState ] = useState( {} );
+
+	useEffect( () => {
+		setPaymentMethodsState(
+			// by default, all the checkboxes should be "checked"
+			availablePaymentMethods
+				.filter( ( method ) =>
+					[ 'giropay', 'sofort', 'sepa_debit' ].includes( method )
+				)
+				.reduce(
+					( map, paymentMethod ) => ( {
+						...map,
+						[ paymentMethod ]: true,
+					} ),
+					{}
+				)
```

Comment:
I like this filter x includes x reduce way of creating an array key of true/false values.